### PR TITLE
Add Graylog Enterprise documentation

### PR DESCRIPTION
**Pull request by bernd**
- Add captions to the navigation sidebar to separate OSS and Enterprise
  docs
- Point Archive installation notes to the Enterprise setup page
- Add Enterprise setup page

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -13,6 +13,7 @@ Contents:
 
 .. toctree::
    :maxdepth: 2
+   :caption: Graylog Server
```
Comment (edmundoa) [inline index.rst]: Maybe call this caption simply Graylog? In the end we describe more than just the server (web interface, and some plugins)

File: pages/enterprise/setup.rst [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      - Directory
+    * - :doc:`/pages/installation/virtual_machine_appliances`
+      - ``/opt/graylog/plugins/``
+    * - :doc:`/pages/installation/operating_system_packages`
+      - ``/usr/share/graylog-server/plugin/``
+    * - :doc:`/pages/installation/manual_setup`
+      - ``/<extracted-graylog-tarball-path>/plugin/``
+
+Make sure to install the enterprise plugin JAR files alongside the other Graylog plugins.
+Your plugin directory should looke similar to this after installing the enterprise plugins.
```
Comment (edmundoa) [inline pages/enterprise/setup.rst]: Typo in `looke`

File: pages/enterprise/setup.rst [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+::
+
+  $ tar -tzf graylog-enterprise-plugins-1.0.0.tgz
+  graylog-enterprise-plugins-1.0.0/plugin/graylog-plugin-archive-1.0.0.jar
+  graylog-enterprise-plugins-1.0.0/plugin/graylog-plugin-license-1.0.0.jar
+
+Depending on the Graylog setup method you have used, you have to install the plugins into different locations.
+
+.. list-table:: Plugin Installation Locations
```
Comment (joschi) [inline pages/enterprise/setup.rst]: This information would also be useful in the normal installation docs.

Comment (edmundoa) [inline pages/enterprise/setup.rst]: We should also indicate that the plugins need to be installed in all Graylog servers inside the cluster. I think that is mentioned somewhere in the docs, but we should remark it here as well.

**Comment by bernd** [comment]
Addresses all comments.

**Comment by bernd** [comment]
Added missing includes directory.

**Comment by edmundoa** [comment]
LGTM 👍

**Pull request merged**